actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -64,6 +64,12 @@ def initialize # :nodoc:
     end
   end
 
+  class KeyError < ArgumentError
```

Comment:
```suggestion
  # Raised when initializing Parameters with keys that's aren't strings or symbols.
  # 
  #   params = ActionController::Parameters.new({ a: "b" } => "c"))
  #   # => ActionController::InvalidKeyError: all keys must be Strings or Symbols
  class InvalidKeyError < ArgumentError
```
I thought about using `TypeError` here, but I don't think it fits the case exactly. A subclass of `ArgumentError` is a better. Calling it `KeyError` is a bit confusing because further up the file we extend `KeyError` above for `ParameterMissing`.

Reply:
Good point thanks. I made the change but took the liberty of naming the error InvalidParameterKey. The name seems to be more clear outside of AC context:

`ActionController::InvalidParameterKey` vs `ActionController::InvalidKeyError`

wdyt?